src/main/java/org/matsim/mosaik2/prepare/NodeMatcher.java

```diff
@@ -0,0 +1,98 @@
+package org.matsim.mosaik2.prepare;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class NodeMatcher {
+
+    Map<String, MatchedLinkID> parseNodeMatching(String filePath) throws IOException {
+
+        Map<String, MatchedLinkID> hashMap = new HashMap<>();
```

Comment:
hash map is not a meaningful name for a variable. If you want to go for something generic please at least change to 'result'